src/main/resources/schemata/metamorph.xsd

```diff
@@ -0,0 +1,927 @@
+            <enumeration value="value"/>
+            <enumeration value="name-value"/>
+          </restriction>
+        </simpleType>
+      </attribute>
+    </complexType>
+  </element>
+  <element name="not-equals">
+    <annotation>
+      <documentation>Returns value only if equality does not hold.
+</documentation>
+    </annotation>
+    <complexType>
+      <attribute name="string" type="string" use="required"/>
+    </complexType>
+  </element>
+  <element name="buffer">
+    <annotation>
+      <documentation>Buffers literals and resleases them on a flush signal. (Used to delay
+litereals)
+</documentation>
+    </annotation>
+    <complexType>
+      <attribute name="flushWith" type="string" use="optional" default="record"/>
+    </complexType>
+  </element>
+  <element name="case">
+    <annotation>
+      <documentation>Upper/lower-case transformation.</documentation>
+    </annotation>
+    <complexType>
+      <attribute name="to" use="required">
+        <simpleType>
+          <restriction base="string">
+            <enumeration value="upper"/>
+            <enumeration value="lower"/>
+          </restriction>
+        </simpleType>
+      </attribute>
+      <attribute name="language" type="string" use="optional"/>
+    </complexType>
+  </element>
+  <element name="constant">
+    <annotation>
+      <documentation>Sets literal value to a constant.</documentation>
+    </annotation>
+    <complexType>
+      <attribute name="value" type="string" use="required"/>
+    </complexType>
+  </element>
+  <element name="script">
+    <annotation>
+      <documentation>Delegates literal processing to a JavaScript
+</documentation>
+    </annotation>
+    <complexType>
+      <attribute name="invoke" type="string" use="required">
+        <annotation>
+          <documentation>Name of JavaScript function.</documentation>
+        </annotation>
+      </attribute>
+      <attribute name="file" type="string" use="required">
+        <annotation>
+          <documentation>JavaScript file.</documentation>
+        </annotation>
+      </attribute>
+    </complexType>
+  </element>
+  <element name="normalize-utf8">
+    <complexType/>
+  </element>
+  <element name="substring">
+    <annotation>
+      <documentation>Returns a substring</documentation>
+    </annotation>
+    <complexType>
+      <attribute name="start" default="0" use="optional">
+        <simpleType>
+          <restriction base="int">
+            <minInclusive value="0"/>
+          </restriction>
+        </simpleType>
+      </attribute>
+      <attribute name="end" use="optional" default="0">
+        <simpleType>
+          <restriction base="int">
+            <minInclusive value="0"/>
+          </restriction>
+        </simpleType>
+      </attribute>
+    </complexType>
+  </element>
+  <element name="urlescape">
+    <annotation>
+      <documentation>URL escaping according to the guava 15.0 UrlEscapers.urlFragmentEscaper library.
```

Comment:
Thanks for explaining the purpose of the XSDs. It seems the documentation above is out of date.

In general this adds a lot of duplication for something that looks like it's essentially a one-line bug fix of URL encoding.

Does the code in your `UrlEscaper` basically do the same thing as `URLEncode` in metafacture-core, but better, i.e. handling a special case we need correctly? If it also handles general URL encoding correctly, maybe we should replace the `URLEncode` implementation in our metafacture-core fork with your implementation.